src/gflownet/algo/config.py

```diff
@@ -121,19 +126,20 @@ class AlgoConfig:
     """
 
     method: str = "TB"
-    global_batch_size: int = 64
+    num_from_policy: int = 64
+    num_from_dataset: int = 0
+    valid_num_from_policy: int = 64
+    valid_num_from_dataset: int = 0
     max_len: int = 128
     max_nodes: int = 128
     max_edges: int = 128
     illegal_action_logreward: float = -100
-    offline_ratio: float = 0.5
-    valid_offline_ratio: float = 1
     train_random_action_prob: float = 0.0
     train_det_after: Optional[int] = None
     valid_random_action_prob: float = 0.0
     sampling_tau: float = 0.0
-    tb: TBConfig = TBConfig()
-    moql: MOQLConfig = MOQLConfig()
-    a2c: A2CConfig = A2CConfig()
-    fm: FMConfig = FMConfig()
-    sql: SQLConfig = SQLConfig()
+    tb: TBConfig = field(default_factory=TBConfig)
+    moql: MOQLConfig = field(default_factory=MOQLConfig)
+    a2c: A2CConfig = field(default_factory=A2CConfig)
+    fm: FMConfig = field(default_factory=FMConfig)
+    sql: SQLConfig = field(default_factory=SQLConfig)
```

Comment:
If you were to create two configs, `a = Config()`, `b=Config()`, doing say `a.algo.max_len = 32` would change both `a` and `b`, because they'd both share `Config.algo == a.algo == b.algo`.